File: Algorithms/Sorts/Java/ShellSort.java
public class ShellSort {
	int sort(int arr[]) {
		for (int gap = arr.length / 2; gap > 0; gap /= 2) {
			for (int i = gap; i < arr.length; i += 1) {
				int temp = arr[i];
				int j;
				for (j = i; j >= gap && arr[j - gap] > temp; j -= gap)
					arr[j] = arr[j - gap];
				arr[j] = temp;
			}
		}
		return 0;
	}

	public static void main(String args[]) {
		int arr[] = { 12, 34, 54, 2, 3 };
		ShellSort shellSort = new ShellSort();
		shellSort.sort(arr);
		System.out.println("The array, after performing shell sort is : ");
		for (int i = 0; i < arr.length; ++i)
			System.out.print(arr[i] + " ");
	}
}

File: Algorithms/Sorts/Csharp/ShellSort.cs
public static class ShellSort
{
    public static void Sort(int[] array)
    {
        for (int gap = array.Length / 2; gap > 0; gap /= 2)
        {
            for (int i = gap; i < array.Length; i++)
            {
                int temp = array[i];
                int j;
                for (j = i; j >= gap && array[j - gap] > temp; j -= gap)
                {
                    array[j] = array[j - gap];
                }

                array[j] = temp;
            }
        }
    }
    public static void Main()
    {
        int[] array = new int[]{ 9, 8, 7, 5, 4, 3, 2, 1};
        
        Console.WriteLine("Unsorted array:");
        PrintArray(array);

        
        Sort(array);
        
        Console.WriteLine("Sorted array:");
        PrintArray(array);

    }

    private static void PrintArray(int[] array)
    {
        for (int i = 0; i < array.Length; i++)
        {
            Console.Write($"{array[i]} ");
        }
        Console.WriteLine(";");
    }
}